framework/src/restart/DataIO.C

```diff
@@ -218,13 +218,11 @@ dataLoad(std::istream & stream, std::string & v, void * /*context*/)
   unsigned int size = 0;
   stream.read((char *) &size, sizeof(size));
 
-  // Read the string
-  char* s = new char[size+1];
-  stream.read(s, sizeof(char)*(size+1));
+  // Resize the string data
+  v.resize(size);
 
-  // Store the string and clean up
-  v = s;
-  delete[] s;
+  // Read the string
+  stream.read(&v[0], sizeof(char) * size);
```

Comment:
Wait, what changed in C++11?  I thought &v[0] is guaranteed to work iff the vector (in this case string?) is non-empty, regardless of standard.  C++11 added http://en.cppreference.com/w/cpp/container/vector/data to make handling the possibly-empty cases easier.